### DISCUSSION TITLE
[Help] Problems with swagger-ui behind a reverse proxy.

### DISCUSSION THREAD

**Discussion by mustafamotiwala** [quarkusio/quarkus#39175] [Q&A] [answered]
Hi,
I have a quarkus app running on say http://example.com/app. However that's behind a K8S Nginx Reverse proxy which maps `/app` to my quarkus app. However when I try to reach the swagger-ui at:
http://example.com/app/q/swagger-ui I get redirected to http://example.com/q/swagger-ui and the swagger-UI doesn't load.
I have in my `application.properties`
```
quarkus.swagger-ui.always-include=true
```

I have also tried mapping it to a different URL using:
```
quarkus.swagger-ui.always-include=true
quarkus.swagger-ui.path=/app/docs
```
But that also didn't work. I simply get a 404 from the quarkus app. If I however do:
```
quarkus.swagger-ui.always-include=true
quarkus.swagger-ui.path=/docs
```
And attempt to hit: http://example.com/app/docs - I am redirected to http://example.com/docs and then I get a 404 from the ingress nginx.

I am using Quarkus 3.8.1
Would appreciate any help to get this to work, or maybe this is a bug?

Thanks.

[a comment by an automated account is not shown]

**Comment by phillip-kruger** [comment]
Are you setting the context root to app ?

`quarkus.http.root-path=/app`

**Reply by mustafamotiwala** [reply]
By setting `quarkus.http.root-path=/app` the quarkus application's web context path becomes `/app` This is a problem because then the path for the app, when coupled with the ingress path becomes:
`http://example.com/app/app`

What I need, is to tell swagger-ui to not attempt redirects. Currently when I do `/app/swagger-ui` I get redirected to `/q/swagger-ui`. I with to have a way where I can control what the external path is for the user instead of what the swagger servlet thinks the external path should be.

If that makes sense?

**Reply by mustafamotiwala** [reply]
Here is an example, of where the swagger-ui request handler is issuing a redirect and causes the prefix to be lost.

![image](https://github.com/quarkusio/quarkus/assets/236083/9d01bcf1-b1ac-4ceb-aaae-64a4ad362a68)

**Reply by phillip-kruger** [reply]
I am not sure I follow properly (a reproducer will help). To change the `/q/` part in quarkus, you can use [quarkus.management.root-path](https://quarkus.io/guides/all-config#quarkus-vertx-http_quarkus-management-root-path)

**Reply by phillip-kruger** [reply]
So do you want swagger ui and your app under /app ?

**Reply by mustafamotiwala** [reply]
Well I'd like to know how I can have the path be preserved. The `302` is causing the front-end reverse proxy to break.

**Reply by mustafamotiwala** [reply]
I am not too fussed about the `q/` path. I can easily change the swagger path using:
`quarkus.swagger-ui.path=/docs`
But then I get the same issue when the `302` is issued, it wrecks the ingress proxy's routing since the path will not be preserved.

**Reply by phillip-kruger** [reply]
I am not sure where the 302 comes from ? As far as I know, we do not do any redirects in swagger ui. I know that if you reference a naked path (so ending on /) we try and redirect to /index.html, but i don't think this is the case here. The only other redirect could be in management (@cescoffier  ?)

**Reply by phillip-kruger** [reply]
Also look at https://quarkus.io/guides/management-interface-reference, this might / might no help

**Reply by cescoffier** [reply]
We do have 302 responses when we hit webjars. Is Swagger a webjar?

**Reply by phillip-kruger** [reply]
Yes it is, the only redirect there that I know about, is when we go from naked ( so root dir) to /index.html.

**Reply by mustafamotiwala** [reply, marked as answer]
So a few things to get this to work. Thanks @phillip-kruger  - your answer clued me in on what is going on. What I found is if I hit the url: `example.com/app/docs` then I get a `302` and get redirected to `example.com/docs/` - and that's where your comment clicked. The trailing `/`. If I hit `example.com/app/docs/` - I get the swagger UI.

I then had to follow up and fix the URLs for the API using `quarkus.smallrye-openapi.servers` and setting it to `%prod.quarkus.smallrye-openapi.servers=/app`

Thanks once again @phillip-kruger & @cescoffier

**Reply by phillip-kruger** [reply]
Great !

**Reply by mustafamotiwala** [reply]
I know this closes the discussion, but do you think this could be considered a bug. i.e. shouldn't issue a `302-Redirect` if the trailing `/` is missing?

**Reply by phillip-kruger** [reply]
@cescoffier ? W.d.y.t?

**Reply by siddharthapd** [reply]
this should be a bug IMHO. even the api docs its pointed to `/q/openapi` instead of `/app/q/openapi`. It should be automatically resolved using servers or some way !
